chaindataloader/sideload.go

```diff
@@ -0,0 +1,73 @@
+package chaindataloader
+
+import (
+	"fmt"
+
+	"github.com/btcsuite/btcd/wire"
+)
+
+const (
+
+	// ErrEndOfFile is returned when we have reached the end of a file while reading it.
+	ErrEndOfFile = sideLoadErr("end of file reached")
```

Comment:
See above re just using `io.EOF`, can even just make an type alias for it.

Reply:
So do we expect that that all implementations of the sideload.Reader would return an `io.EOF` when if reaches its end of file. I guess the custom error was an attempt to hide implementation details from the block manager and have one error returned in spite of.

Reply:
> So do we expect that that all implementations of the sideload.Reader would return an io.EOF when it reaches its end? 

I think either that, or can continue to use `ErrEndOfFile`. Just that `io.EOF` is something standardized in Go.